src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1055,7 +1066,8 @@ private IOperation CreateBoundAssignmentOperatorOrMemberInitializerOperation(Bou
         }
 
         private static bool IsMemberInitializer(BoundAssignmentOperator boundAssignmentOperator) =>
-            boundAssignmentOperator.Right?.Kind == BoundKind.ObjectInitializerExpression || boundAssignmentOperator.Right?.Kind == BoundKind.CollectionInitializerExpression;
+            boundAssignmentOperator.Right?.Kind == BoundKind.ObjectInitializerExpression || boundAssignmentOperator.Right?.Kind == BoundKind.CollectionInitializerExpression ||
+            boundAssignmentOperator.Right?.Kind == BoundKind.DynamicObjectInitializerMember;
```

Comment:
>boundAssignmentOperator.Right?.Kind == BoundKind.DynamicObjectInitializerMember [](start = 12, length = 79)

I am not sure about the purpose of this condition, it looks like binder always puts it on the left side of an assignment